src/main/java/com/ibm/devops/connect/CRPipeline/UploadASoCTestResult.java

```diff
@@ -0,0 +1,230 @@
+                    int highFindings = (int)getHighCountMethod.invoke(action);
+                    int infoFindings = (int)getInfoCountMethod.invoke(action);
+
+                    // The ASoC plugin has a bug where the info count is 0.  That is not accurate, so we adjust
+                    if(infoFindings == 0 && totalFindings != (lowFindings + mediumFindings + highFindings + infoFindings)) {
+                        infoFindings = totalFindings - (lowFindings + mediumFindings + highFindings);
+                    }
+
+                    listener.getLogger().println("Total ----> " + totalFindings);
+                    listener.getLogger().println("info ----> " + infoFindings);
+                    listener.getLogger().println("low ----> " + lowFindings);
+                    listener.getLogger().println("medium ----> " + mediumFindings);
+                    listener.getLogger().println("high ----> " + highFindings);
+
+                    JSONObject payload = new JSONObject();
+                    JSONObject application = new JSONObject();
+                    JSONObject record = new JSONObject();
+                    JSONObject value = new JSONObject();
+                    JSONObject buildObj = new JSONObject();
+
+                    buildObj.put("url", this.buildUrl);
+
+                    application.put("id", this.appId);
+                    application.put("name", this.appName);
+                    application.put("externalId", this.appExtId);
+
+                    value.put("High", highFindings);
+                    value.put("Medium", mediumFindings);
+                    value.put("Low", lowFindings);
+                    value.put("Info", infoFindings);
+
+                    record.put("metricDefinitionId", this.metricDefinition);
+                    record.put("dataFormat", "json");
+                    record.put("recordName", this.recordName);
+                    record.put("pluginType", "templatePlugin");
+                    record.put("value", value);
+
+                    payload.put("dataSet", "AppScan on Cloud Scan Results");
+                    payload.put("environment", this.environment);
+                    payload.put("tenantId", this.tenantId);
+                    payload.put("record", record);
+                    payload.put("application", application);
+                    payload.put("build", buildObj);
+
+
+                    listener.getLogger().println("Uploading Payload Doc");
+                    try {
+                        CloudPublisher.uploadQualityDataRaw(payload.toString());
+                    } catch (Exception ex) {
+                        listener.error("Error uploading metric data: " + ex.getClass() + " - " + ex.getMessage());
+                        build.setResult(Result.FAILURE);
+                    }
+
+                    listener.getLogger().println("UPLOADED PAYLOAD: " + payload.toString());
```

Comment:
I would put the payload printout before uploading. That way if there is an error, its easy to see what was attempted to be uploaded